From 123560457eb40606cba196037cf4db00c0116adc Mon Sep 17 00:00:00 2001
From: Nicolas Boichat <drinkcat@google.com>
Date: Tue, 28 Jun 2016 15:58:42 +0800
Subject: [PATCH] pvrimage: Do not recompute strides, no matter the number of
 planes.

When using an EGL image where width != stride, Android's
dri2_create_image_android_native_buffer computes the correct
strides, there is no need to recompute it (incorrectly...).

BUG=chrome-os-partner:54834
BUG=b:29059384
TEST=On oak-cheets, play CGP Grey video at 240p.

Change-Id: Idc05983e9312aae4e0b444ca965ccabef76f764d
---
 src/mesa/drivers/dri/pvr/pvrimage.c | 12 +-----------
 1 file changed, 1 insertion(+), 11 deletions(-)

diff --git a/src/mesa/drivers/dri/pvr/pvrimage.c b/src/mesa/drivers/dri/pvr/pvrimage.c
index f912cc4..22e5249 100644
--- a/src/mesa/drivers/dri/pvr/pvrimage.c
+++ b/src/mesa/drivers/dri/pvr/pvrimage.c
@@ -976,7 +976,6 @@ __DRIimage *PVRDRICreateImageFromDmaBufs(__DRIscreen *screen,
                                          void *loaderPrivate)
 {
 	__DRIimage *image;
-	int iStride;
 
 	image = CommonImageSetup(loaderPrivate);
 	if (!image)
@@ -996,17 +995,8 @@ __DRIimage *PVRDRICreateImageFromDmaBufs(__DRIscreen *screen,
 		return NULL;
 	}
 
-	if (image->psShared->psFormat->uiNumPlanes == 1)
-	{
-		iStride = strides[0];
-	}
-	else
-	{
-		iStride = width * PVRDRIPixFmtGetBlockSize(image->psShared->psFormat->eIMGPixelFormat);
-	}
-
 	image->psEGLImage = PVRDRIEGLImageCreateFromBuffers(width, height,
-							    iStride,
+							    strides[0],
 							    image->psShared->psFormat->eIMGPixelFormat,
 							    image->psShared->eColourSpace,
 							    image->psShared->eChromaUInterp,
-- 
2.8.0.rc3.226.g39d4020

